src/opnsense/mvc/app/models/OPNsense/Unbound/Migrations/M1_0_4.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * Copyright (C) 2022 Deciso B.V.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\Unbound\Migrations;
+
+use OPNsense\Base\BaseModelMigration;
+use OPNsense\Core\Backend;
+
+class M1_0_4 extends BaseModelMigration
+{
+    public function post($model)
+    {
+        /* ensure an existing blocklist is migrated during version upgrade */
+        if (!file_exists('/var/unbound/data/dnsbl.json')) {
```

Comment:
if this happens on a major upgrade it might not happen at all or stall the install... :(

Reply:
I don't have another suggestion apart from informing users to manually reapply their blocklist settings. In any case, some form of migration is necessary

Reply:
We can add an empty dnsbl.json sample file in unbound.opnsense.d also because we shouldn't anchor the file origin inside the chroot as we might want to clear it on each reboot perhaps. /var MFS is gone but I think for chroots we should treat it as if volatile.
Then we could add a late dnsbl call to rc.syshook "start" facility if we are concerned about "timely" updates.

Reply:
FWIW, base migration starts prior to bootstrapping the chroot for unbound in a new install / image.

Reply:
@fichtner https://github.com/opnsense/core/pull/6083/commits/9205a0fd868aba9a58fea6ecfb3785aea2d2efd7